src/services/users.js

```diff
@@ -0,0 +1,191 @@
+const { User } = require('../models');
+const { UserRoles } = require('../models');
+const { UserStatuses } = require('../models');
+const bcrypt = require('bcryptjs');
+
+class UsersService {
+  async createUser (req, res) {
+    try {
+      const userExist = await this.checkUserExists(req);
+      if (!userExist) {
+        const roleUser = await this.getRole('User');
+        const newUser = await this.createBasicUser(req);
+        newUser.role = roleUser._id;
+        await newUser.save();
+        return newUser ? res.json(true) : res.boom.notFound();
+      } else {
+        res.boom.badRequest('user already exists');
+      }
+    } catch (err) {
+      res.boom.badRequest();
+    }
+  }
+
+  async createAdmin (req, res) {
+    try {
+      const userExist = await this.checkUserExists(req);
+      if (!userExist) {
+        const roleAdmin = await this.getRole('Admin');
+        const newAdmin = await this.createBasicUser(req);
+        newAdmin.role = roleAdmin._id;
+        await newAdmin.save();
+        return newAdmin ? res.json(true) : res.boom.notFound();
+      } else {
+        res.boom.badRequest('user already exists');
+      }
+    } catch (err) {
+      res.boom.badRequest();
+    }
+  }
+
+  async getUsers (req, res) {
+    const { _id, name, email, status, role, skip, take } =
+      (typeof req.query !== 'undefined' && req.query) || {};
+    const query = {};
+    if (_id != null) query._id = _id;
```

Comment:
No necessity to write if (_id != null)... You can just write if(_id)... Remove these nulls

Reply:
Fixed